cmd/cloudtrail/permission-denied.go

```diff
@@ -0,0 +1,163 @@
+package cloudtrail
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	ctUtil "github.com/openshift/osdctl/cmd/cloudtrail/pkg"
+	ctAws "github.com/openshift/osdctl/cmd/cloudtrail/pkg/aws"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var (
+	permissionDeniedErrorStr = ".*Client.UnauthorizedOperation.*"
+)
+
+type permissionDeniedEventsOptions struct {
+	ClusterID string
+	StartTime string
+	PrintOptions
+	QueryOptions
+}
+
+type PrintOptions struct {
+	PrintUrl bool
+	PrintRaw bool
+}
+
+type QueryOptions struct {
+	StartTime time.Time
+	ReadOnly  bool
+}
```

Comment:
**chore:** You're not using those (here and in write-events)? Is this implementation finished?